lighthouse-core/gather/gather-runner.js

```diff
@@ -214,26 +221,55 @@ class GatherRunner {
         const artifacts = Object.assign({}, tracingData);
         passes.forEach(pass => {
           pass.gatherers.forEach(gatherer => {
+            if (typeof gatherer.artifact === 'undefined') {
```

Comment:
I don't think I want a gatherer to return undefined as it's easily mistakable for a failure to set `artifact`. If you want to say error then a -1 would do it.